nni/algorithms/compression/v2/pytorch/utils/config_validation.py

```diff
@@ -1,8 +1,40 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
+from logging import Logger
+from typing import Dict, List
 from schema import Schema, And, SchemaError
 
+from torch.nn import Module
+
+
+class CompressorSchema:
+    def __init__(self, data_schema: List[Dict], model: Module, logger: Logger):
+        assert isinstance(data_schema, list)
+        self.data_schema = data_schema
+        self.compressor_schema = Schema(self._modify_schema(data_schema, model, logger))
+
+    def _modify_schema(self, data_schema: List[Dict], model: Module, logger: Logger) -> List[Dict]:
+        if not data_schema:
+            return data_schema
+
+        for sub_schema in data_schema:
+            for k, old_schema in sub_schema.items():
+                if k == 'op_types' or (isinstance(k, Schema) and k._schema == 'op_types'):
+                    new_schema = And(old_schema, lambda n: validate_op_types(model, n, logger))
+                    sub_schema[k] = new_schema
+                if k == 'op_names' or (isinstance(k, Schema) and k._schema == 'op_names'):
+                    new_schema = And(old_schema, lambda n: validate_op_names(model, n, logger))
+                    sub_schema[k] = new_schema
+
+            sub_schema = And(data_schema[0], lambda d: validate_op_types_op_names(d))
```

Comment:
I think this line might not modify the object in `data_schema`. Is that a correct behavior? Not sure because I haven't worked a lot with `Schema`

Reply:
yes, it's a bug, fixed it.